diff --git a/climate/version.py b/climate/version.py
--- a/climate/version.py
+++ b/climate/version.py
@@ -13,6 +13,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from pbr import version
+class VersionInfo(object):
+    release = "REDHATBLAZARRELEASE"
+    version = "REDHATBLAZARVERSION"
 
-version_info = version.VersionInfo('climate')
+    def version_string(self):
+        return self.version
+
+    def release_string(self):
+        return self.release
+
+
+version_info = VersionInfo()
